inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningRecommender.java

```diff
@@ -174,36 +187,48 @@ private boolean isAlreadyInCleanList(List<AnnotationObject> cleanRecommendationL
         return flag;
     }
 
-    private void removeRejectedOrSkippedAnnotations()
+    private void removeRejectedOrSkippedAnnotations(boolean filterSkippedRecommendation,
+        Date learnSkippedRecommendationTime)
     {
         List<LearningRecord> records = learningRecordService.getAllRecordsByDocumentAndUserAndLayer
             (annotatorState.getDocument(), annotatorState.getUser().getUsername(), selectedLayer);
         for (List<AnnotationObject> recommendations: listOfRecommendationsForEachToken) {
             recommendations.removeIf(recommendation -> doesContainRejectedOrSkippedRecord(records,
-                recommendation));
+                recommendation, filterSkippedRecommendation, learnSkippedRecommendationTime));
         }
         listOfRecommendationsForEachToken.removeIf(recommendationsList ->
             recommendationsList.size() == 0);
     }
 
     private boolean doesContainRejectedOrSkippedRecord(List<LearningRecord> records,
-                                                       AnnotationObject aRecommendation)
+        AnnotationObject aRecommendation, boolean filterSkippedRecommendation,
+        Date learnSkippedRecommendationTime)
     {
         boolean flag = false;
         for (LearningRecord record: records) {
-            if ((record.getUserAction().equals("rejected") || record.getUserAction().equals
-                ("skipped")) && record.getSourceDocument().getName()
-                .equals(aRecommendation.getDocumentName()) && record.getOffsetTokenBegin() ==
-                aRecommendation.getOffset().getBeginToken() && record.getOffsetTokenEnd() ==
-                aRecommendation.getOffset().getEndToken() && record.getAnnotation().equals
-                (aRecommendation.getAnnotation())) {
+            if ((record.getUserAction().equals("rejected") || (
+                record.getUserAction().equals("skipped") && filterSkippedRecommendation)
+                && needFilterByTime(learnSkippedRecommendationTime, record)) && record
+                .getSourceDocument().getName().equals(aRecommendation.getDocumentName())
+                && record.getOffsetTokenBegin() == aRecommendation.getOffset().getBeginToken()
+                && record.getOffsetTokenEnd() == aRecommendation.getOffset().getEndToken() && record
+                .getAnnotation().equals(aRecommendation.getAnnotation())) {
                 flag = true;
                 break;
             }
         }
         return flag;
     }
 
+    private boolean needFilterByTime (Date learnSkippedTime, LearningRecord record) {
+        if (learnSkippedTime != null && learnSkippedTime.compareTo(record.getActionDate()) == 1) {
```

Comment:
Could be written as 

```
return learnSkippedTime != null && learnSkippedTime.compareTo(record.getActionDate()) == 1;
```

But I am not sure what this code is actually doing. A comment would be nice. Also, reordering might allow for skipping the null check, e.g. if it is guaranteed that `record.getActionDate()` always returns non-null.

Reply:
have added a comment and set ActionDate nullable = false